xls/examples/lfsr_proc.x

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+////////////////////////////////////////////////////////////////////////////////
+// 8-bit LFSR proc
+// 
+// Two operations are supported:
+// 1. setting the seed
+// 2. getting the next value from the LFSR counter
+////////////////////////////////////////////////////////////////////////////////
+
+import xls.examples.lfsr
+
+proc user_module {
+  output_s: chan<u8> out;
+  seed_r: chan<u8> in;
+
+  init {
+    (u8:1)
+  }
+
+  config(output_s: chan<u8> out, seed_r: chan<u8> in) {
+    (output_s, seed_r)
+  }
+
+  next(tok: token, state: u8) {
+    let (tok, seed, seed_valid) = recv_non_blocking(tok, seed_r, state);
+    let state = if(seed_valid) {
+        (seed)
```

Comment:
actually I'm not sure if you need this, as the `recv_non_blocking` will default to `state` if there is nothing to `recv`?
you could simply `send(seed)` and return `lfsr::lfsr` as the new state.

wdyt?

Reply:
Ah yes, you are right. I simplified the code.